qiskit/transpiler/layout.py

```diff
@@ -738,3 +743,218 @@ def final_virtual_layout(self, filter_ancillas: bool = True) -> Layout:
         res = self.final_index_layout(filter_ancillas=filter_ancillas)
         pos_to_virt = {v: k for k, v in self.input_qubit_mapping.items()}
         return Layout({pos_to_virt[index]: phys for index, phys in enumerate(res)})
+
+    @classmethod
+    def from_property_set(
+        cls, dag: DAGCircuit, property_set: PropertySet
+    ) -> TranspileLayout | None:
+        """Construct the :class:`TranspileLayout` by reading out the fields from the given
+        :class:`.PropertySet`.  Returns ``None`` if there are no layout-setting keys present.
+
+        This includes combining the different keys of the property set into the full set of initial
+        and final layouts, including virtual permutations.
+
+        This does not invalidate or in any way mutate the given property set.  In order to
+        "canonicalize" the property set afterwards, call :meth:`write_into_property_set`.
+
+        This reads the following property-set keys:
+
+        ``layout``
+            **Required**. The :class:`.Layout` object mapping virtual qubits (potentially expanded
+            with ancillas) to physical-qubit indices.  This corresponds directly to
+            :attr:`initial_layout`.
+
+            .. note::
+                In all standard use, this is a required field.  However, if
+                ``virtual_permutation_layout`` is set, then a "trivial" layout will be inferred,
+                even if the circuit is not actually laid out to hardware.  This is an unfortunate
+                limitation of the circuit data model, where it is not possible to specify a final
+                permutation without also having an initial layout.
+
+        ``original_qubit_indices``
+            **Required** (but automatically set by the :class:`.PassManager`).  The mapping
```

Comment:
Just checking this resolves to `qiskit.transpiler.PassManager` and not `qiskit.passmanager.PassManager`. I assume it does because of the reverse resolution order. But figured I'd ask because it's the kind of thing that's easy to slip.

Reply:
The one in `qiskit.passmanager` is called `BasePassManager`.

Reply:
Sphinx doesn't let you use the prefix-`.` lookup if there's more than one possible match - it's a warning/error (either way, it fails our build).